R/cache.R

```diff
@@ -96,30 +89,41 @@ initialize_cache = function(cache_dir) {
 #
 # @param type
 # The type that is downloaded, not really necessary
-cached = function(fun, type, id, ..., cache_dir = FALSE) {
+cached = function(fun, type, id, test_server, parquet = FALSE, ..., cache_dir = FALSE) {
   if (isFALSE(cache_dir)) {
     return(fun(id, ...))
   }
 
-  path = file.path(cache_dir, type)
-  file = file.path(path, sprintf("%i.qs", id))
+  path = file.path(cache_dir, ifelse(test_server, "test", "public"), type)
+  file = file.path(path, sprintf("%i.%s", id, ifelse(parquet, "parquet", "qs")))
 
   if (file.exists(file)) {
-    lg$debug("Loading object from cache", type = type, id = id, file = file)
-    obj = try(qs::qread(file, nthreads = getOption("Ncpus", 1L)))
-    if (!inherits(obj, "try-error")) {
-      return(obj)
+    if (parquet) {
+      lg$debug("Returning parquet path.", type = type, id = id, file = file)
+      return(file)
+    } else {
+      lg$debug("Loading object from cache", type = type, id = id, file = file)
+      obj = try(qs::qread(file, nthreads = getOption("Ncpus", 1L)))
+      if (!inherits(obj, "try-error")) {
+        return(obj)
+      }
+      lg$debug("Failed to retrieve object from cache", type = type, id = id, file = file)
     }
-    lg$debug("Failed to retrieve object from cache", type = type, id = id, file = file)
   }
 
   if (!dir.exists(path)) {
     dir.create(path, recursive = TRUE)
   }
 
   obj = fun(id, ...)
-  lg$debug("Storing object in cache", type = type, id = id, file = file)
-  qs::qsave(obj, file = file, nthreads = getOption("Ncpus", 1L))
+  if (parquet) {
+    lg$debug("Moving parquet data from tempfile to cache.", type = type, id = id, file = file)
+    file.copy(obj, file, overwrite = TRUE)
```

Comment:
this is not a move, this is a copy

why can't we download the file to the cache dir directly?

Reply:
But I also unlink the object afterwards. Does this physically copy the data the way I did it? So I use `file.rename()`? 

This does not coincide with the way the `cached()` function works. The function is being wrapped by `cached()` does not know that it is being cached, i.e. it does not know the cache directory.